openmicroscopy/openmicroscopy
# Viewer dataset

# What this PR does

When you open the image viewer for an image in a Dataset, we add ```?dataset=id``` to the URL so that the viewer knows which Dataset the image is in.

Although we don't use this info in the current viewer, if you configure the default viewer to be iViewer

```$ bin/omero config set omero.web.viewer.view omero_iviewer.views.index```

this will use the Dataset ID to load thumbnails:

![screen shot 2017-03-28 at 16 35 32](https://cloud.githubusercontent.com/assets/900055/24413853/9c0eb0ce-13d4-11e7-82fa-39e8e44fad2f.png)

Have also discussed pathviewer using this - cc @chris-allan.

# Testing this PR

1. Open a viewer by:
    - Double-clicking in tree
    - Double-clicking thumbnail in Dataset
    - Right-click on thumbnail in Dataset -> Open Link in New Tab
    - Click "Full Viewer" button in General tab
2. Viewer should open as normal, with URL ending with ```?dataset={id}```
3. Images in other containers (Tags, Orphaned, Shares) should render thumbnails OK and work as before.
